rules/internal/framework_middleman.bzl

```diff
@@ -94,7 +99,18 @@ def _framework_middleman(ctx):
         )
         providers.append(resource_provider)
 
-    return providers
+    # Just to populate the extension safety provider from `rules_apple`.
+    partial_output = partial.call(
+        partials.extension_safe_validation_partial(
+            is_extension_safe = ctx.attr.extension_safe,
+            rule_label = ctx.label,
+            # Pass 'ctx.attr.framework_deps' once 'partials.extension_safe_validation_partial'
+            # is populated on from 'apple_framework_packaging' implementation.
+            targets_to_validate = [],
```

Comment:
Yep we might need some 1-2 tweaks to a make sure a framework shard amongst an app and ext have the same flags. e.g:

```
App -> A.framework
Ext -> A.framework 
```
```
App.app/Frameworks/A.framework
App.app/Plugins/Ext.appex
```

I think this should work fine with some minor tweaks 👍

Reply:
Will add tests for extension + app packaging and hopefully validation using `targets_to_validate` in a follow up! Thanks, Jerry!